Library/Homebrew/formula_installer.rb

```diff
@@ -312,8 +318,24 @@ def prelude
       end
     end
 
+    # Needs to be done before expand_dependencies for compute_dependencies
+    fetch_bottle_tab if pour_bottle?
+
+    @ran_prelude_fetch = true
+  end
+
+  sig { void }
+  def prelude
+    prelude_fetch unless @ran_prelude_fetch
```

Comment:
Maybe check for `@ran_prelude_fetch` at the start of `#prelude_fetch` as a guard instead in case it picks up other callers.